includes/amp-helper-functions.php

```diff
@@ -318,6 +318,133 @@ function amp_add_frontend_actions() {
 	add_action( 'wp_head', 'amp_add_amphtml_link' );
 }
 
+/**
+ * Determine whether AMP is available for the current URL.
+ *
+ * @todo This should go into the AmpProject\AmpWP namespace?
+ * @since 1.6
+ *
+ * @return bool Whether there is an AMP version for the provided URL.
+ * @global string $pagenow
+ * @global WP_Query $wp_query
+ */
+function is_amp_available() {
+	global $pagenow, $wp_query;
+
+	// Short-circuit for admin requests or requests to non-frontend pages.
+	if ( is_admin() || in_array( $pagenow, [ 'wp-login.php', 'wp-signup.php', 'wp-activate.php' ], true ) ) {
+		return false;
+	}
+
+	$warned        = false;
+	$error_message = sprintf(
+		/* translators: %1$s: is_amp_endpoint(), %2$s: the current action, %3$s: the wp action, %4$s: the WP_Query class, %5$s: the amp_skip_post() function */
+		__( '%1$s was called too early and so it will not work properly. WordPress is currently doing the "%2$s" action. Calling this function before the "%3$s" action means it will not have access to %4$s and the queried object to determine if it is an AMP response, thus neither the "%5$s" filter nor the AMP enabled toggle will be considered.', 'amp' ),
```

Comment:
Seeing this message on the login screen.

<img width="490" alt="Screen Shot 2020-06-25 at 10 47 02 AM" src="https://user-images.githubusercontent.com/9020968/85752783-50e7a580-b6d1-11ea-86a5-983565e78c50.png">

Reply:
Fixed in e1d740f3bae9ce490c53435e09a03bae64eab763